test/e2e/suite/nodeadm/nodeadm_test.go

```diff
@@ -98,21 +106,26 @@ var _ = Describe("Hybrid Nodes", func() {
 						k8sVersion = test.OverrideNodeK8sVersion
 					}
 
-					testNode := test.NewTestNode(ctx, instanceName, nodeName, k8sVersion, nodeOS, provider, e2e.Large, e2e.CPUInstance)
+					remoteCommandRunner := ssm.NewStandardLinuxSSHOnSSMCommandRunner(test.SSMClient, test.JumpboxInstanceId, test.Logger)
+					logCollector := peered.StandardLinuxLogCollector{
+						Runner: remoteCommandRunner,
+					}
+					testNode := test.NewTestNode(ctx, instanceName, nodeName, k8sVersion, nodeOS, provider, e2e.Large, e2e.CPUInstance, remoteCommandRunner, logCollector)
 					Expect(testNode.Start(ctx)).To(Succeed(), "node should start successfully")
 					Expect(testNode.Verify(ctx)).To(Succeed(), "node should be fully functional")
 
 					test.Logger.Info("Testing Pod Identity add-on functionality")
-					verifyPodIdentityAddon := test.NewVerifyPodIdentityAddon(testNode.PeerdNode().Name)
+					verifyPodIdentityAddon := test.NewVerifyPodIdentityAddon(testNode.PeeredInstance().Name)
 					Expect(verifyPodIdentityAddon.Run(ctx)).To(Succeed(), "pod identity add-on should be created successfully")
 
 					test.Logger.Info("Resetting hybrid node...")
-					n := testNode.PeerdNode()
+					i := testNode.PeeredInstance()
 					cleanNode := test.NewCleanNode(
 						provider,
-						testNode.PeeredNode.NodeInfrastructureCleaner(*n),
-						n.Name,
-						n.Instance.IP,
+						testNode.PeeredNode.NodeInfrastructureCleaner(*i),
+						i.Name,
+						i.IP,
+						remoteCommandRunner,
```

Comment:
nit: you could consider adding options to this method to override the command runner without adding another argument, not a blocker